### PR TITLE
core: support inline file transfers

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

File: src/Simplex/Chat/Protocol.hs [hunk 1]
```diff
@@ -59,18 +66,66 @@ updateEntityConnStatus connEntity connStatus = case connEntity of
...
 
+instance StrEncoding AppMessageBinary where
+  strEncode AppMessageBinary {tag, msgId, body} = smpEncode (tag, Tail body)
+  -- strEncode AppMessageBinary {tag, msgId, body} = smpEncode (tag, msgId', body)
+    where
+      msgId' = maybe B.empty (\(SharedMsgId mId') -> mId') msgId
+  strP = do
+    (tag, Tail body) <- smpP
+    -- (tag, msgId', body) <- smpP
+    -- let msgId = if B.null msgId' then Nothing else Just (SharedMsgId msgId')
```
Comment (jr-simplex) [inline src/Simplex/Chat/Protocol.hs:126]:
```suggestion
```

File: tests/ChatTests.hs [hunk 2]
```diff
@@ -66,15 +71,17 @@ chatTests = do
...
+    describe "send and receive file" $ fileTestMatrix2 runTestFileTransfer
+    it "send and receive file inline (without accepting)" testInlineFileTransfer
+    describe "send and receive a small file" $ fileTestMatrix2 runTestSmallFileTransfer
+    describe "sender cancelled file transfer before transfer" $ fileTestMatrix2 runTestFileSndCancelBeforeTransfer
     it "sender cancelled file transfer during transfer" testFileSndCancelDuringTransfer
     it "recipient cancelled file transfer" testFileRcvCancel
-    it "send and receive file to group" testGroupFileTransfer
-    it "sender cancelled group file transfer before transfer" testGroupFileSndCancelBeforeTransfer
+    describe "send and receive file to group" $ fileTestMatrix3 runTestGroupFileTransfer
+    it "send and receive file to group (without accepting)" testInlineGroupFileTransfer
```
Comment (jr-simplex) [inline tests/ChatTests.hs:81]:
```suggestion
    it "send and receive file inline to group (without accepting)" testInlineGroupFileTransfer
```

File: tests/ChatTests.hs [hunk 3]
```diff
@@ -119,7 +126,7 @@ chatTests = do
   describe "maintenance mode" $ do
     it "start/stop/export/import chat" testMaintenanceMode
     it "export/import chat with files" testMaintenanceModeWithFiles
-    it "encrypt/decrypt database" testDatabaseEncryption
+    xit "encrypt/decrypt database" testDatabaseEncryption
```
Comment (jr-simplex) [inline tests/ChatTests.hs:129]: why disable this test?

Comment (epoberezkin) [reply tests/ChatTests.hs:129]: accidentally

**Pull request merged**